src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java

```diff
@@ -80,49 +89,86 @@ public void initialize() {
 
         portId = (String) getConfig().get(CONFIGURATION_PORT);
 
-        if (portId == null || portId.length() == 0) {
+        if (portId.isEmpty()) {
             logger.debug("ZWave port is not set.");
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
                     ZWaveBindingConstants.OFFLINE_PORT_UNSET);
             return;
         }
 
         super.initialize();
-        logger.info("Connecting to serial port '{}'", portId);
+        watchdog = scheduler.scheduleWithFixedDelay(this::watchSerialPort, WATCHDOG_INIT_SECONDS,
+                WATCHDOG_CHECK_SECONDS, TimeUnit.SECONDS);
+
+    }
+
+    /**
+     * Gets a serial port identifier for a given name.
+     * Workaround for getIdentifier in SerialPortManager class, because it doesn't detected correctly that the device is
+     * unplugged.
+     *
+     * @param the name
+     * @return a serial port identifier or null
+     */
+    private @Nullable SerialPortIdentifier getSerialPortIdentifier(final String name) {
+        Optional<SerialPortIdentifier> opt = serialPortManager.getIdentifiers().filter(id -> id.getName().equals(name))
+                .findFirst();
+        if (opt.isPresent()) {
+            return opt.get();
+        } else {
+            return null;
+        }
+    }
+
+    private void watchSerialPort() {
         try {
-            SerialPortIdentifier portIdentifier = serialPortManager.getIdentifier(portId);
+            SerialPortIdentifier portIdentifier = getSerialPortIdentifier(portId);
             if (portIdentifier == null) {
```

Comment:
I suggest that we make the port a required configuration in the thing XML file and then we can remove this check.